feat: subscription size action

MrMuzyk opened the pull request:
<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->

### Details
<!-- Explanation of the change or anything fishy that is going on -->

Added navigation action when pressing `subscription size` button

### Fixed Issues
<!---
1. Please postfix `$` with a URL link to the GitHub issue this Pull Request is fixing. For example, `$ https://github.com/Expensify/App/issues/<issueID>`.
2. Please postfix  `PROPOSAL:` with a URL link to your GitHub comment, which contains the approved proposal (i.e. the proposal that was approved by Expensify).  For example, `PROPOSAL: https://github.com/Expensify/App/issues/<issueID>#issuecomment-1369752925`

Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the github issue and your comment proposal ; otherwise, the linking and its automation will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/<issueID>
$ https://github.com/Expensify/App/issues/<issueID(comment)>

Do NOT only link the issue number like this: $ #<issueID>
--->
$ https://github.com/Expensify/App/issues/38625
PROPOSAL:


### Tests
<!---
Add a numbered list of manual tests you performed that validates your changes work on all platforms, and that there are no regressions present.
Add any additional test steps if test steps are unique to a particular platform.
Manual test steps should be written so that your reviewer can repeat and verify one or more expected outcomes in the development environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

- go to subscription settings page
- press on subscription size button
- [x] Verify that no errors appear in the JS console

### Offline tests
<!---
Add any relevant steps that validate your changes work as expected in a variety of network states e.g. "offline", "spotty connection", "slow internet", etc. Manual test steps should be written so that your reviewer and QA testers can repeat and verify one or more expected outcomes. If you are unsure how the behavior should work ask for advice in the `#expensify-open-source` Slack channel.
--->

### QA Steps
<!---
Add a numbered list of manual tests that can be performed by our QA engineers on the staging environment to validate that your changes work on all platforms, and that there are no regressions present.
Add any additional QA steps if test steps are unique to a particular platform.
Manual test steps should be written so that the QA engineer can repeat and verify one or more expected outcomes in the staging environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

No QA
- [x] Verify that no errors appear in the JS console

### PR Author Checklist
<!--
This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged!
-->

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

<!-- add screenshots or videos here -->
n/a
</details>

<details>
<summary>Android: mWeb Chrome</summary>

<!-- add screenshots or videos here -->

https://github.com/Expensify/App/assets/23487612/794c76b9-d4be-4e23-8eee-14cb665d264f


</details>

<details>
<summary>iOS: Native</summary>

<!-- add screenshots or videos here -->
n/a
</details>

<details>
<summary>iOS: mWeb Safari</summary>

<!-- add screenshots or videos here -->

https://github.com/Expensify/App/assets/23487612/16aac406-9097-4661-a460-becd700b39f6


</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

<!-- add screenshots or videos here -->


https://github.com/Expensify/App/assets/23487612/134d6ab8-db66-4399-8658-de638309f838

https://github.com/Expensify/App/assets/23487612/5ff83db1-bdba-4ecd-bba7-291946d8e0bc


</details>

<details>
<summary>MacOS: Desktop</summary>

<!-- add screenshots or videos here -->
n/a
</details>

MrMuzyk committed:
feat: subscription size action

allroundexperts commented:
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Android: mWeb Chrome</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>iOS: Native</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>iOS: mWeb Safari</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>MacOS: Desktop</summary>

<!-- add screenshots or videos here -->

</details>

trjExpensify commented:
@allroundexperts assigned you!

allroundexperts commented:
@trjExpensify Is this change for web only? On Android, I'm getting the following:

https://github.com/Expensify/App/assets/30054992/c5c2af4e-f7b1-4a9a-8201-20771896678a

trjExpensify commented:
> @trjExpensify Is this change for web only? On Android, I'm getting the following:
> 
>  Screen.Recording.2024-06-05.at.7.44.57.PM.mov

Yeah, due to the Google and Apple overlords we can't let you manage your subscription on native iOS or Android. 🙄

allroundexperts commented on src/pages/settings/Subscription/SubscriptionDetails/index.tsx:
```diff
@@ -44,33 +46,30 @@ function SubscriptionDetails() {
-                style={styles.mt5}
-            />
-        ) : (
+    // This section is only shown when the subscription is annual
```
This section is visible even if the subscription type is not annual:

https://github.com/Expensify/App/assets/30054992/620d9741-67c1-4b03-bd33-a5312e7a106e

allroundexperts replied:
Ah, I guess it does not show up if the set subscription is not annual.

MrMuzyk replied:
Condition was checking it against data coming straight from the API. Because change is only happening on FE now (API call will be implemented in the next phase) it was not working as expected. I've updated the condition to check it against the selectedOption that we store inside state of the component. It should work as expected now.

trjExpensify commented:
> @trjExpensify Is this change for web only? On Android, I'm getting the following:
> 
>  Screen.Recording.2024-06-05.at.7.44.57.PM.mov

On this though. If you have a subscription size set, it should be read-only and visible on native. Not sure where @MrMuzyk is at with that one, but here for illustration: 

<img width="412" alt="image" src="https://github.com/Expensify/App/assets/16232057/2a508d6d-3081-4fbb-9a55-01a646a78853">

allroundexperts commented:
> > @trjExpensify Is this change for web only? On Android, I'm getting the following:
> > Screen.Recording.2024-06-05.at.7.44.57.PM.mov
> 
> On this though. If you have a subscription size set, it should be read-only and visible on native. Not sure where @MrMuzyk is at with that one, but here for illustration:
> 
> <img alt="image" width="412" src="https://private-user-images.githubusercontent.com/16232057/336893023-2a508d6d-3081-4fbb-9a55-01a646a78853.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc1OTk5MjAsIm5iZiI6MTcxNzU5OTYyMCwicGF0aCI6Ii8xNjIzMjA1Ny8zMzY4OTMwMjMtMmE1MDhkNmQtMzA4MS00ZmJiLTlhNTUtMDFhNjQ2YTc4ODUzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjA1VDE1MDAyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFjOWUxNDQ5Nzk0ODVkNzQ3MzM4YjhiMzUzMDk3ZGU3NGE2MDFkMjQxM2Y4YjE1MzgwZDNjNWY5NWJkY2VmNmYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.K0y3LSBfNnsO4KDm4yJk639u7rqFwTS8IVJf39pC01A">

I think my subscription size is set to 0. That is why its not showing, I think.

allroundexperts reviewed (approved):
Changes are looking good. One small comment. Also, I'm assuming that this is design approved?

dannymcclain commented:
Can't seem to get this running locally (it's always something 😅) but I have a few questions about the videos/screenshots. It looks like this PR is specific to the subscription size RHP, so please forgive me if any of these are being handled elsewhere!

In this one, why don't these cards have a max-width?
![CleanShot 2024-06-05 at 10 41 53@2x](https://github.com/Expensify/App/assets/7516624/eafc84b4-2110-4106-a9b0-52cb7c4fa11f)

I believe the button on this second screen should be `Save`, not `Next`, as I think at this point you're committing your changes.
![CleanShot 2024-06-05 at 10 44 05@2x](https://github.com/Expensify/App/assets/7516624/989c3fcf-53b7-4167-bfc0-8d013b810954)

These radios do not look properly styled in light mode—this looks like the dark mode styles.
![CleanShot 2024-06-05 at 10 46 44@2x](https://github.com/Expensify/App/assets/7516624/8dd9a162-14ff-494d-8a52-12f0af4b98a4)

They should look like this in light mode:
![image](https://github.com/Expensify/App/assets/7516624/5db67345-7539-4a60-9696-3f47b968e27d)

Super tiny, but we need a `:` at the end of the top sentence here. It should read `Confirm your new annual subscription details:`
![CleanShot 2024-06-05 at 10 48 37@2x](https://github.com/Expensify/App/assets/7516624/7b3f2d7e-ab50-4118-b3c6-5e6255ecdcca)

shawnborton commented:
Good catch on all of those Danny!

For the radio buttons, don't those already exist as a component and they work already across light and dark modes? I hope we aren't rewriting them!

MrMuzyk committed:
fix: change condition for displaying size section

MrMuzyk committed:
fix: cr fixes

MrMuzyk commented:
>In this one, why don't these cards have a max-width?

We must've missed that, adding it now.

<img width="1920" alt="image" src="https://github.com/Expensify/App/assets/23487612/45b07693-7b59-4db5-8df1-663373bf607c">

----

>I believe the button on this second screen should be Save, not Next, as I think at this point you're committing your changes

>Super tiny, but we need a : at the end of the top sentence here. It should read Confirm your new annual subscription details:

I've assumed its a mistake and did set it to `Save`. Also added that `:`

<img width="1920" alt="image" src="https://github.com/Expensify/App/assets/23487612/ffee7648-a51c-4476-ad1c-5cac402f8137">

----

>These radios do not look properly styled in light mode—this looks like the dark mode styles.

>For the radio buttons, don't those already exist as a component and they work already across light and dark modes? I hope we aren't rewriting them!

Fixed them, we had one unnecessary style added to them. Should look fine now

<img width="1920" alt="image" src="https://github.com/Expensify/App/assets/23487612/1b29c0d4-7a2f-4eb8-a608-d137c12a8472">

JKobrynski commented on src/components/OptionsPicker/OptionItem.tsx:
```diff
@@ -54,10 +54,7 @@ function OptionItem({title, icon, onPress, isSelected = false, isDisabled, style
-                                    isChecked={isSelected}
-                                    selectCircleStyles={styles.sectionSelectCircle}
-                                />
+                                <SelectCircle isChecked={isSelected} />
```
Have you checked if it looks like it should on both themes? E.g like this for dark theme 

![Screenshot 2024-06-06 at 14 45 57](https://github.com/Expensify/App/assets/44364829/4950c49e-c8d2-47a1-b1cc-135a3da9fe53)

I haven't ran your code but I made that change on my branch and the background is too dark

MrMuzyk replied:
I wasn't aware that it's too dark now. I've reverted it to default state that should work for both platforms. Changing it again so its fine on both themes

MrMuzyk committed:
Merge branch 'main' of https://github.com/Expensify/App into feat/subscription-size-action

MrMuzyk committed:
fix: changed select circle bg

MrMuzyk commented:
@shawnborton I've updated PR description with a few videos. I didn't include native as this button is not even visible on these platforms

dannymcclain commented:
@shawnborton an interesting thing is that in Figma, our radio-checkboxes do not have a background color/fill. So when we put them on the `highlight-background` card, all is well. It looks like here they _do_ have a background fill that is set to `app-background` so they stand out a bit more than they do in Figma. 

You can see what I'm talking about here:
![CleanShot 2024-06-06 at 09 10 05@2x](https://github.com/Expensify/App/assets/7516624/d391e58a-f86b-40b5-a847-e561e4ce9f84)
![CleanShot 2024-06-06 at 09 10 22@2x](https://github.com/Expensify/App/assets/7516624/47cf7ff3-47d0-431b-9021-1f31affb331f)

Do you think this is something worth addressing? I honestly don't think they look bad with the background fill, as it does kinda make them stand out a bit more, which I don't really think is a bad thing.

MrMuzyk commented:
@dannymcclain It was pointed out to me by @JKobrynski that in figma circles actually had different background as usual so I've updated it and recorded the videos. In the screenshot I've sent earlier I still had background that was a bit darker on them. Let me know which version is better and should be implemented :)

dannymcclain commented:
@MrMuzyk Ah perfect! The ones in the videos you recorded look great.

dannymcclain commented:
Oh great catch Shawn. You're saying it should look like this:

![CleanShot 2024-06-06 at 09 51 37@2x](https://github.com/Expensify/App/assets/7516624/20260eb3-9fa1-45bb-ac63-3c00acfeb379)

Can that subscription size input exist "unset" / with no value? Might need to tag in @MitchExpensify and @trjExpensify to answer that one. I see in the video you can't set it to `0`, so does that mean it always needs to have a value of at least `1`?

shawnborton commented:
Nice, that's what I was thinking - I feel like if a value does exist, it should be reflected in that push row.

trjExpensify commented:
>Can that subscription size input exist "unset" / with no value? Might need to tag in @MitchExpensify and @trjExpensify to answer that one. I see in the video you can't set it to 0, so does that mean it always needs to have a value of at least 1?

You can only set a subscription of `1` or higher and you can't change it to `0`. That said, I think since [this PR](https://github.com/Expensify/Auth/pull/10119/files), the subscription size can exist "unset" at `0` if it's before they've taken a billable action. In which case, it would appear as `0` until they set a subscription size, or take a billable action and we set it to `1` (minimally, could be more if there are multiple actors). @flodnv as the PR author can perhaps double confirm.

dannymcclain commented:
Ok, I think I understand... But I think there's a difference between `unset` and `set to 0`. So we need to figure out how this push input should be displayed:

![image](https://github.com/Expensify/App/assets/7516624/9077b32e-b0cd-4464-bb67-fb6c4ab35579)

In my mind, `unset` means that subscription size has NO value—the value is essentially null/undefined/non-existent. `Set to 0` means the subscription size has a value and that value is 0. Have I just made this worse/more confusing?

trjExpensify commented:
>In my mind, unset means that subscription size has NO value—the value is essentially null/undefined/non-existent

I think a user will think this way as well, personally. Especially as you can't opt-in for a subscription size of `0`.

MrMuzyk commented:
I agree that there is a difference between unset value and 0. At the moment it was set to 0. I can change it to an empty string so it is displayed different. Does it look better?

https://github.com/Expensify/App/assets/23487612/91e17efc-048a-4f80-8475-218c5d5e9f1e

MrMuzyk committed:
Merge branch 'main' of https://github.com/Expensify/App into feat/subscription-size-action

trjExpensify commented:
I think that looks better to me! @shawnborton @dannymcclain?

flodnv commented:
>You can only set a subscription of 1 or higher and you can't change it to 0. That said, I think since [this PR](https://github.com/Expensify/Auth/pull/10119/files), the subscription size can exist "unset" at 0 if it's before they've taken a billable action. In which case, it would appear as 0 until they set a subscription size, or take a billable action and we set it to 1 (minimally, could be more if there are multiple actors). @flodnv as the PR author can perhaps double confirm.

IIRC it's possible that the existing value is `0` (or unset, which is the same as `0`), however a user cannot _set it_ to `0`.

dannymcclain commented:
>I think that looks better to me!

That looks better to me too. The user can open the RHP and decide they don't want to mess with anything and just close it without issue. I think it's good now.

I am ignoring Flo's comment about unset being the same as being set to 0 😂. I think for _this purpose_, because a user cannot set the value to 0, we can treat it as unset (like what's shown in the updated video) if no specific value from the user has been given.

MrMuzyk committed:
fix: set default size to empty string

allroundexperts commented:
Looking fine to me:

https://github.com/Expensify/App/assets/30054992/b9184cb6-78e7-4b9f-8e0f-74afb41d9d38

https://github.com/Expensify/App/assets/30054992/ae558bf3-18b0-4c87-a4db-40360c5390cf

allroundexperts commented:
@dannymcclain can confirm though from design perspective.

shawnborton commented:
Hmm @allroundexperts in your videos, given that a subscription size exists once you open the RHP, I would think the push row would show the subscription size no?

dannymcclain commented:
Looking pretty good except in your first video, when you open the subscription size RHP it shows that it has a value of `10`. Which means the push input should look like this:
![CleanShot 2024-06-07 at 11 08 40@2x](https://github.com/Expensify/App/assets/7516624/3c323ae1-d51f-40ab-8de7-e91c4c354ed3)

But it's showing as empty instead. Also I think that input should be auto-focused when the RHP opens, right?

Your second video looks good, except the radio options should be stacked for that viewport width, though I'm almost positive that is being handled elsewhere already.

allroundexperts commented:
> Hmm @allroundexperts in your videos, given that a subscription size exists once you open the RHP, I would think the push row would show the subscription size no?

I think that's not the case. As far as I've understood, those are draft values of the form. Draft values don't show up in the push row.

MrMuzyk commented:
>Hmm @allroundexperts in your videos, given that a subscription size exists once you open the RHP, I would think the push row would show the subscription size no?

These are draft values being displayed there.

>But it's showing as empty instead. Also I think that input should be auto-focused when the RHP opens, right?

Changing it

MrMuzyk committed:
Merge branch 'main' of https://github.com/Expensify/App into feat/subscription-size-action

MrMuzyk committed:
fix: added autoFocus to size input

MrMuzyk commented:
@amyevans Can you review this so we can merge it?

MrMuzyk committed:
Merge branch 'main' of https://github.com/Expensify/App into feat/subscription-size-action

amyevans reviewed (approved):
Looks good! I'm going to remove the QA steps though since this isn't a fully functional feature at this point

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/9473255300) to staging by https://github.com/amyevans in version: 1.4.82-0 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/9502360613) to production by https://github.com/mountiny in version: 1.4.82-4 🚀

platform | result
---|---
🤖 android 🤖|failure ❌
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅
🕸 web 🕸|success ✅
